aa_stripe/models.py

```diff
@@ -462,17 +469,17 @@ class StripeWebhook(models.Model):
 
     def _parse_coupon_notification(self, action):
         coupon_id = self.raw_data["data"]["object"]["id"]
+        created = timestamp_to_timezone_aware_date(self.raw_data["data"]["object"]["created"])
         if action == "created":
-            try:
-                StripeCoupon(coupon_id=coupon_id).save(force_retrieve=True)
-            except stripe.error.InvalidRequestError:
-                # do not fail in case the coupon has already been removed from Stripe before we received the webhook
-                pass
+            if StripeCoupon.objects.filter(coupon_id=coupon_id, created=created).exists():
+                raise StripeCouponAlreadyExists
+
+            StripeCoupon(coupon_id=coupon_id).save(force_retrieve=True)
         elif action == "updated":
-            StripeCoupon.objects.filter(coupon_id=coupon_id, is_deleted=False).update(
+            StripeCoupon.objects.filter(coupon_id=coupon_id, created=created, is_deleted=False).update(
```

Comment:
This will not trigger post_save on coupon object. Is this intended? If so, why?

Reply:
Yes, it is intentional. There is no point in calling the API again in the `save()` method just to update object's metadata.

Reply:
Are you sure? What if the system uses metadata to trigger some behavior and system is listening to post_save signal? or pre or whatever?

Reply:
Good point. I added `super(StripeCoupon, coupon).save()` to execute the post save without the need to call Stripe API.